turbopack/crates/turbo-tasks/src/manager.rs

```diff
@@ -1414,13 +1469,42 @@ impl<B: Backend + 'static> TurboTasksBackendApi<B> for TurboTasks<B> {
     }
 }
 
-pub(crate) fn current_task(from: &str) -> TaskId {
+pub(crate) fn current_task_if_available(from: &str) -> Option<TaskId> {
     match CURRENT_TASK_STATE.try_with(|ts| ts.read().unwrap().task_id) {
         Ok(id) => id,
-        Err(_) => panic!("{from} can only be used in the context of turbo_tasks task execution"),
+        Err(_) => panic!(
+            "{from} can only be used in the context of a turbo_tasks task execution or \
+             turbo_tasks run"
+        ),
+    }
+}
+
+pub(crate) fn current_task(from: &str) -> TaskId {
+    match CURRENT_TASK_STATE.try_with(|ts| ts.read().unwrap().task_id) {
+        Ok(Some(id)) => id,
+        Ok(None) | Err(_) => {
+            panic!("{from} can only be used in the context of a turbo_tasks task execution")
+        }
     }
 }
 
+pub async fn run<T: Send + 'static>(
+    tt: Arc<dyn TurboTasksApi>,
+    future: impl Future<Output = Result<T>> + Send + 'static,
+) -> Result<T> {
+    let (tx, rx) = tokio::sync::oneshot::channel();
+
+    tt.run(Box::pin(async move {
+        let result = future.await?;
+        tx.send(result)
+            .map_err(|_| anyhow!("unable to send result"))?;
```

Comment:
Looks like this is just copying `run_once`, so this is fine, but ideally it'd turn the error into a string and wrap it in `TurboTasksError`: https://github.com/vercel/next.js/blob/e60a57fcae78f69f3fe26521f8896cc8a29e49db/turbopack/crates/turbo-tasks/src/backend.rs#L428C12-L428C27